### PR TITLE
Feature/orchestrator

### PR DESCRIPTION
**Pull request by blankdots**
- Add SDA orchestrator chart.
- Add test for deployment of SDA-Orchestrator

### CODE REVIEW

File: sda-orch/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          value: {{ .Values.db.port | quote }}
+        - name: DB_SSLMODE
+          value: {{ .Values.db.sslMode | quote }}
+        - name: BROKER_HOST
+          value: {{ required "A valid MQ Host is required" .Values.broker.host | quote }}
+        - name: BROKER_PORT
+          value: {{ .Values.broker.port | quote }}
+        - name: BROKER_VHOST
+          value: {{ .Values.broker.vhost | quote }}
+        - name: BRO KER_EXCHANGE
```
Comment (jbygdell) [inline sda-orch/templates/deployment.yaml:105]:
```suggestion
        - name: BROKER_EXCHANGE
```

File: sda-orch/values.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+
+image:
+  repository: neicnordic/sda-orch
+  tag: latest
+  pullPolicy: Always
+
```
Comment (jbygdell) [inline sda-orch/values.yaml:6]:
```suggestion
logLevel: "info"

```

File: sda-orch/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        - name: tls
+          mountPath: /tls
+    {{- end }}
+      containers:
+      - name: sda-orchestrate
+        image: "{{ .Values.image.repository }}:{{ .Values.image.imageTag }}"
+        imagePullPolicy: {{ .Values.image.imagePullPolicy | quote }}
+        securityContext:
+          allowPrivilegeEscalation: false
+        env:
```
Comment (jbygdell) [inline sda-orch/templates/deployment.yaml:60]:
```suggestion
        env:
        - name: DEFAULT_LOG
          value: {{ .Values.LogLevel | quote }}
```

File: sda-orch/values.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  queue:
+    inbox: ""
+    completed: ""
+    verify: ""
+
+
+resources:
+  requests:
+    memory: "128Mi"
+    cpu: "100m"
```
Comment (jbygdell) [inline sda-orch/values.yaml:67]: Is it really this power hungry?

**Review by jbygdell — CHANGES_REQUESTED** [review]

File: sda-orch/values.yaml (hunk 4 above)
Comment (blankdots) [reply sda-orch/values.yaml:67]: not really, i think could be lower, there are just 3 processes that pass info around. I will update.

Comment (blankdots) [reply sda-orch/values.yaml:68]: this seems to be the limit on other services as well ... I would rather keep it consistent

**Review by jbygdell — APPROVED** [review]

File: sda-orch/templates/_helpers.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+{{- end -}}
+
+{{- define "tlsPath" -}}
+{{- ternary "/tls/certs" .Values.tlsPath (empty .Values.tlsPath) -}}
+{{- end -}}
+
+
+{{/**/}}
+{{- define "dbOutUser" -}}
+{{- ternary "lega_out" .Values.db.userOutgest (empty .Values.db.userOutgest) -}}
```
Comment (pontus) [inline sda-orch/templates/_helpers.yaml:40]: I know e.g. sda-svc writes these in the same way, but is there a reason for not just doing `default "lega_out" .Values.db.userOutgest`?

File: sda-orch/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+Edit the values.yaml file and specify the relevant parts of the `config` section.  
+
+Parameter | Description | Default
+--------- | ----------- | -------
+`global.tlsPath` | Default TLS path for certs and key in the pod. | `/tls/certs`
+`revisionHistory` | Number of revisions to keep for the option to rollback a deployment | `3`
+`podAnnotations` | Annotations applied to pods of all services. |`{}`
+`pkiService` | If an external PKI infrastructure is used set this to true. |`false`
+`rbacEnabled` | Use role based access control. |`true`
```
Comment (pontus) [inline sda-orch/README.md:13]: pkiPermissions?

File: sda-orch/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# SDA Orchestrator Service
+
+## Installing the Chart
+
+Edit the values.yaml file and specify the relevant parts of the `config` section.  
+
+Parameter | Description | Default
+--------- | ----------- | -------
+`global.tlsPath` | Default TLS path for certs and key in the pod. | `/tls/certs`
```
Comment (pontus) [inline sda-orch/README.md:9]: Just tlsPath. (And image and logLevel?)

File: sda-orch/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+          value: {{ .Values.broker.queue.complete | quote }}
+        - name: INBOX_QUEUE
+          value: {{ .Values.broker.queue.inbox | quote }}
+        - name: FILES_QUEUE
+          value: {{ .Values.broker.queue.files | quote }}
+        - name: STABLEIDS_QUEUE
+          value: {{ .Values.broker.queue.stableid | quote }}
+        {{- if .Values.sslmode.ssl }}
+        - name: SSL_CACERT
+          value: {{ include "tlsPath" . }}/ca.crt
```
Comment (pontus) [inline sda-orch/templates/deployment.yaml:118]: I still haven't fully grasped this, are we certain quote isn't needed for these?

**Review by pontus — APPROVED** [review]

File: sda-orch/templates/networkpolicies.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: sda-orch
+  namespace: {{ .Release.Namespace }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "sda.fullname" . }}
+  policyTypes:
+  - Ingress
```
[the remark this answers is not in the document]
Comment (blankdots) [reply sda-orch/templates/networkpolicies.yaml:41]: yes, i need to test this :( will fix

File: sda-orch/templates/podsecuritypolicy.yaml [hunk 10]
```diff
@@ -0,0 +1,40 @@
+{{- if and .Values.rbacEnabled .Values.podSecurityPolicy.create }}
+apiVersion: extensions/v1beta1
```
[the remark this answers is not in the document]
Comment (blankdots) [reply sda-orch/templates/podsecuritypolicy.yaml:2]: yes, did not fully test this. just used what was in the other charts as well. so probably more fixes coming

File: sda-orch/templates/_helpers.yaml (hunk 5 above)
Comment (blankdots) [reply sda-orch/templates/_helpers.yaml:40]: true noticed that, will simplify it. I though i would need more complex checks

File: sda-orch/templates/deployment.yaml (hunk 8 above)
Comment (blankdots) [reply sda-orch/templates/deployment.yaml:118]: reused what was in other charts and in my tests seems to work

File: sda-orch/README.md (hunk 7 above)
Comment (blankdots) [reply sda-orch/README.md:9]: forgotten, my bad. will fix

File: sda-orch/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+        emptyDir: {}
+      - name: {{ ternary "tls" "tls-certs" (empty .Values.pkiPermissions) }}
+        secret:
+          defaultMode: 0440
+          secretName: {{ template "sda.fullname" . }}-orchestrate-certs
+      {{- if .Values.pkiPermissions }}
+      - name: tls
+        emptyDir:
+          medium: Memory
+          sizeLimit: 10Mi
```
[the remark this answers is not in the document]
Comment (blankdots) [reply sda-orch/templates/deployment.yaml:146]: I followed what was in other charts as well.

File: sda-orch/templates/podsecuritypolicy.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (blankdots) [reply sda-orch/templates/podsecuritypolicy.yaml:2]: I see this is added in https://github.com/neicnordic/sda-helm/pull/27

File: sda-orch/templates/deployment.yaml (hunk 8 above)
Comment (blankdots) [reply sda-orch/templates/deployment.yaml:112]: seems like we have multiple variants of that:

- https://github.com/neicnordic/sda-helm/blob/master/sda-svc/templates/auth-deploy.yaml#L117-L120
- https://github.com/neicnordic/sda-helm/blob/master/sda-svc/templates/ingest-deploy.yaml#L129-L132

I suggest we address this in a different PR.

File: sda-orch/templates/deployment.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (blankdots) [reply sda-orch/templates/deployment.yaml:140]: I see other templates use the same: https://github.com/neicnordic/sda-helm/blob/master/sda-svc/templates/ingest-deploy.yaml#L249-L254 so I recommend we keep as is.

**Pull request merged**